core/math/randombase.h

```diff
@@ -0,0 +1,29 @@
+#ifndef RANDBASE_CORE_H
+#define RANDBASE_CORE_H
+
+#include "math.h"
+#include "math_defs.h"
+#include "thirdparty/misc/pcg.h"
+
+class RandomBase {
+    pcg32_random_t pcg;
+
+public:
+	static const uint64_t DEFAULT_SEED = 12047754176567800795ULL;
+	static const uint64_t DEFAULT_INC = PCG_DEFAULT_INC_64;
+	static const uint64_t RANDOM_MAX = 4294967295;
+
+	RandomBase(uint64_t seed = DEFAULT_SEED, uint64_t inc = PCG_DEFAULT_INC_64);
+
+    void seed(uint64_t x);
+	void randomize();
+	uint32_t rand();
+	_ALWAYS_INLINE_ double randf() { return (double)rand() / (double)RANDOM_MAX; }
```

Comment:
Why randf process double and randd float ? A mistake ?

Reply:
Now that you point that out, I have no idea. [This was simply copy pasted from `math_funcs.h`](https://github.com/Zirak/godot/blob/2d08552af5d85b75652f63aeeb824fe11a94da89/core/math/math_funcs.h#L238), so make it a double mistake? Seems like nobody actually uses `randd` anywhere in Godot, so it could be safe to remove.

Reply:
The best solution would be to fix it and switch it around. Change the functions in `math_funcs.h` too.